File: samples/java/app/src/main/java/org/remdev/wetweexapp/domain/services/TextGenerator.java
package org.remdev.wetweexapp.domain.services;

import org.remdev.wetweexapp.domain.model.GeneratedText;

import java.io.ByteArrayOutputStream;
import java.util.Random;

public class TextGenerator {

    public GeneratedText generateText(Long millis) {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        final long end = System.currentTimeMillis() + millis;
        while (System.currentTimeMillis() < end) {
            baos.write((byte) new Random().nextInt(127));
            try {
                Thread.sleep(10);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        return new GeneratedText(new String(baos.toByteArray()));
    }
}

File: samples/kotlin/app/src/main/java/org/remdev/wetweexapp/domain/services/TextGenerator.kt
package org.remdev.wetweexapp.domain.services

import org.remdev.wetweexapp.domain.model.GeneratedText
import kotlin.random.Random

class TextGenerator {

    fun generateText(millis: Long): GeneratedText {
        val list = mutableListOf<Byte>()
        val end = System.currentTimeMillis() + millis
        while (System.currentTimeMillis() < end) {
            list.add(Random.nextInt(0, 127).toByte())
            Thread.sleep(10)
        }
        return GeneratedText(String(list.toByteArray()))
    }
}